File: potd-gfg/Add 1 to a number represented as linked list (10th January 2022)/Add1ToLinkedList.java
Problem Link:
https://practice.geeksforgeeks.org/problems/add-1-to-a-number-represented-as-linked-list/1

public static Node addOne(Node head) 
    { 
        int carry=addRec(head);
        if(carry==1){
            Node newHead=new Node(1);
            newHead.next=head;
            return newHead;
        }
        return head;
    }
    
    public static int addRec(Node node){
        if(node==null){
            return 1;
        }
        
        int sum=node.data+addRec(node.next);
        node.data=sum%10;
        return sum/10;
    }

File: potd-gfg/Add 1 to a number represented as linked list (10th January 2022)/Add1ToLinkedList.py
#User function Template for python3

'''

class Node:
    def __init__(self, data):   # data -> value stored in node
        self.data = data
        self.next = None
'''

class Solution:
    def addOne(self,head):
        #Returns new head of linked List.
        num = ""
        temp = head
        while head.next != None:
            num += str(head.data)
            head = head.next
        num += str(head.data)
        num = int(num) + 1

        temp.next = None
        temp.data = num
        return temp
